taskcluster/scriptworker_taskgraph/transforms/python_version.py

```diff
@@ -42,37 +42,38 @@ def _resolve_replace_string(item, field, subs):
 
 
 @transforms.add
-def set_script_name(config, jobs):
-    for job in jobs:
-        job.setdefault("attributes", {}).update(
+def set_script_name(config, tasks):
+    for task in tasks:
+        task.setdefault("attributes", {}).update(
             {
-                "script-name": job["name"],
+                "script-name": task["name"],
             }
         )
-        yield job
+        yield task
 
 
 @transforms.add
-def tasks_per_python_version(config, jobs):
+def tasks_per_python_version(config, tasks):
     fields = [
         "description",
         "docker-repo",
         "run.command",
         "worker.command",
         "worker.docker-image",
     ]
-    for job in jobs:
-        for python_version in job.pop("python-versions"):
-            task = deepcopy(job)
-            subs = {"name": job["name"], "python_version": python_version}
+    for task_raw in tasks:
+        task = deepcopy(task_raw)
```

Comment:
Moving the deepcopy outside the loop doesn't seem safe.  AFAICT `_resolve_replace_string` would end up making changes to `task` in the first iteration that persist for later ones.
E.g. https://firefox-ci-tc.services.mozilla.com/tasks/NW_4tQxDSLqsr28eOp1azg is supposed to be using python 3.9 but ends up with 3.8.